### ISSUE TITLE
add context menu component

### ISSUE THREAD

**Issue by christianvogt** [patternfly/patternfly-react#3351]
**Describe the issue. What is the expected and unexpected behavior?**
Topology component has a need to bring up a context menu:
![image](https://user-images.githubusercontent.com/14068621/69552737-fafd2480-0f6c-11ea-833f-22e9582305d2.png)

- Ability to open a context menu popover a specific x, y coordinate identified by the mouse click event.
- Popover should open to the bottom right of the specified coordinate.
- Popver should flip and orient itself accordingly to open and remain visible on screen.
- Menu should receive focus on open.
- Menu should support keyboard navigation through the list of menu actions.
- Menu should trap focus.
- Popover should request close on escape, click outside, or action selected.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**
Our workaround has been very involved. We first created a `Popper` component to handle positioning much like tippy.js. Then we had to mimic the dropdown structure and override some styles to make it work. The end result works great however it would be nice to see a standard popover component in patternfly that's generic for these situations as well as a generic menu component that doesn't require hacking around the dropdown menu.

```
<Popper
      {...other}
      closeOnEsc
      closeOnOutsideClick
      open={isOpen}
      onRequestClose={handleOnRequestClose}
    >
      <DropdownContext.Provider
        value={{
          onSelect: handleOnRequestClose,
          toggleTextClass: styles.dropdownToggleText,
          toggleIconClass: styles.dropdownToggleIcon,
          menuClass: styles.dropdownMenu,
          itemClass: styles.dropdownMenuItem,
          toggleClass: styles.dropdownToggle,
          baseClass: styles.dropdown,
          baseComponent: 'div',
          sectionClass: styles.dropdownGroup,
          sectionTitleClass: styles.dropdownGroupTitle,
          sectionComponent: 'section',
          disabledClass: styles.modifiers.disabled,
          hoverClass: styles.modifiers.hover,
          separatorClass: styles.dropdownSeparator,
        }}
      >
        <div className="pf-c-dropdown pf-m-expanded">
          <DropdownMenu className="pf-c-dropdown__menu topology-context-menu" autoFocus>
            {children}
          </DropdownMenu>
        </div>
      </DropdownContext.Provider>
    </Popper>
```

**What is your product and what release version are you targeting?**
OpenShift web console 4.4

cc @jeff-phillips-18

**Comment by rachael-phillips** [comment]
@mcarrano @mceledonia making you aware of this from a design perspective. What are your thoughts on approach?

**Comment by christianvogt** [comment]
Some thoughts:
I think it would be useful to have a `Menu` and `MenuItem` components that are reusable in places like `Dropdown`. Also `Popover` should be more generic to support opening without an arrow and it should be used with `Dropdown` as well because there are issues there too. Then this same `Popover` can be used for the context menu along with a nested `Menu`.

**Comment by mcarrano** [comment]
@rachael-phillips I agree that this would be a useful component and with @christianvogt 's idea about creating reusable components for elements like menus and menu items. It would help drive consistency in look and behavior between a growing set of things. This is something that has been discussed in the past but we need to figure out timing for the refactoring it would require.

**Comment by mceledonia** [comment]
I like the approach, to me the key here is ensuring we're offering the additional flexibility without too much room for inconsistencies. It seems like these changes would overall drive more consistency to the components, which sounds great to me. I'm not super well versed in the implementation side of how these overlap and interact, so I would defer to someone with more knowledge there.

**Comment by mcoker** [comment]
Agreed, I think this would be valuable and help drive consistency while making the components that use this type of menu currently easier to maintain. We have discussed this in the past and have an issue in the core repo from February that outlines this idea - https://github.com/patternfly/patternfly-next/issues/1429, including creating separate [menu and menu item components](https://github.com/patternfly/patternfly-next/issues/1429#issuecomment-464219629).

**Comment by christianvogt** [comment]
I believe we are also exploring the need for sub menus.
Please keep this in mind when designing the menus and react components.
